src/Infrastructure/LeanCode.ForceUpdate/LeanCode.ForceUpdate.Services/VersionHandler.cs

```diff
@@ -0,0 +1,52 @@
+using Microsoft.AspNetCore.Http;
+
+namespace LeanCode.ForceUpdate.Services;
+
+public class VersionHandler
+{
+    private readonly AndroidVersionsConfiguration androidConfiguration;
+    private readonly IOSVersionsConfiguration iOSConfiguration;
+
+    public VersionHandler(AndroidVersionsConfiguration androidConfiguration, IOSVersionsConfiguration iOSConfiguration)
+    {
+        this.androidConfiguration = androidConfiguration;
+        this.iOSConfiguration = iOSConfiguration;
+    }
+
+    public virtual Task<VersionSupportResult> CheckVersionAsync(Version version, Platform platform, HttpContext context)
+    {
+        var (minimumRequiredVersion, currentlySupportedVersion) = platform switch
+        {
+            Platform.Android
+                => (androidConfiguration.MinimumRequiredVersion, androidConfiguration.CurrentlySupportedVersion),
+            Platform.IOS => (iOSConfiguration.MinimumRequiredVersion, iOSConfiguration.CurrentlySupportedVersion),
+            _ => throw new InvalidOperationException($"Invalid platform: {platform}."),
+        };
+
+        if (version < minimumRequiredVersion)
+        {
+            return Task.FromResult(VersionSupportResult.UpdateRequired);
+        }
+        else if (version >= minimumRequiredVersion && version < currentlySupportedVersion)
+        {
+            return Task.FromResult(VersionSupportResult.UpdateSuggested);
+        }
+        else
+        {
+            return Task.FromResult(VersionSupportResult.UpToDate);
+        }
+    }
+}
+
+public enum Platform
```

Comment:
We don't need to duplicate the `enum`s, we don't gain anything for having them in contracts and here. Let's use normal DTOs.